### PR TITLE
Enable PRC by default

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This PR enables PlanResourceChange by default in the bridge. A bunch of tests had to be fixed up to work with PRC.

fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/1785

### CODE REVIEW

File: pkg/tfbridge/tests/provider_test.go [hunk 1]
```diff
@@ -285,444 +284,6 @@ func TestReproMinimalDiffCycle(t *testing.T) {
 	}`)
 }
 
-func TestValidateInputsPanic(t *testing.T) {
```
Comment (VenelinMartinov) [inline pkg/tfbridge/tests/provider_test.go:288]: Replaced with https://github.com/pulumi/pulumi-gcp/pull/2389 for now and will follow-up in https://github.com/pulumi/pulumi-terraform-bridge/issues/2418

The tests here only test the required behaviour indirectly through panics which no longer happen.

Comment (t0yv0) [reply pkg/tfbridge/tests/provider_test.go:288]: It'd be nice if we had reliable coverage metrics to see that removing these tests doesn't decrease coverage %.

File: pkg/tfbridge/schema_test.go [hunk 2]
```diff
@@ -2808,7 +2797,8 @@ func TestExtractSchemaInputsNestedMaxItemsOne(t *testing.T) {
 				},
 			},
 			expectedOutputs: resource.PropertyMap{
-				"id": resource.NewProperty("MyID"),
+				"id":     resource.NewProperty("MyID"),
+				"__meta": resource.NewStringProperty("{\"schema_version\":\"0\"}"),
```
Comment (t0yv0) [inline pkg/tfbridge/schema_test.go:2801]: We can throw a ticket somewhere that the bridge shouldn't inject annoying `"__meta": resource.NewStringProperty("{\"schema_version\":\"0\"}"` that are equal to the default value.

File: pkg/tfbridge/diff_test.go [hunk 3]
```diff
@@ -1463,24 +1466,6 @@ func TestNestedComputedSetUpdate(t *testing.T) {
 		pulumirpc.DiffResponse_DIFF_SOME)
 }
 
-func TestNestedComputedSetAdd(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (t0yv0) [reply pkg/tfbridge/diff_test.go:1466]: Can we double-check this one? It's not that the state is unknown, it's the Set is passed `[unk]`. 

```func diffTest(t *testing.T, tfs map[string]*v2Schema.Schema, inputs, state map[string]interface{},```

File: pkg/tfbridge/diff_test.go [hunk 4]
```diff
@@ -1538,24 +1523,6 @@ func TestNestedComputedSetIntUpdateReplace(t *testing.T) {
 		pulumirpc.DiffResponse_DIFF_SOME)
 }
 
-func TestNestedComputedSetIntAdd(t *testing.T) {
```
Comment (t0yv0) [inline pkg/tfbridge/diff_test.go:1541]: Same here, can we double-check why it's removed?

**Comment by t0yv0** [comment]
Thanks for cleaning up the test suite! Mostly looking good, I had a few questions on the removed tests.

File: pkg/tfbridge/tests/provider_test.go (hunk 1 above)
Comment (VenelinMartinov) [reply pkg/tfbridge/tests/provider_test.go:288]: It likely does. https://github.com/pulumi/pulumi-gcp/pull/2389 gives some coverage here. I plan to add back tests for the type checking warnings once I solve https://github.com/pulumi/pulumi-terraform-bridge/issues/2418

File: pkg/tfbridge/diff_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (VenelinMartinov) [reply pkg/tfbridge/diff_test.go]: So both diff tests have the same problem - they are unit tests that test only the `makeDetailedDiff` function.

I've added an integration test for the same: https://github.com/pulumi/pulumi-terraform-bridge/pull/2380/files#diff-9c7bc852f0c771404295c8deca79b1819534b5929d97f7d2bd677a2bb5559f44R4089

There is indeed an issue with PRC - it seems like makeDetailedDiff under PRC does not show the diff null -> [Unknown] in the set property. I'll open a separate issue for that.

[the remark this answers is not in the document]
Comment (t0yv0) [reply pkg/tfbridge/diff_test.go]: Can you please leave the tests in, cross-link the issue to why these are skipped, and add it to the Accurate Previews epic. I'd like these tests to be on master. IF they move to an integration test later that's great.

File: pkg/tfbridge/schema_test.go (hunk 2 above)
Comment (VenelinMartinov) [reply pkg/tfbridge/schema_test.go:2801]: https://github.com/pulumi/pulumi-terraform-bridge/issues/2428

File: pkg/tfbridge/diff_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (VenelinMartinov) [reply pkg/tfbridge/diff_test.go]: opened https://github.com/pulumi/pulumi-terraform-bridge/issues/2428 to follow up here

File: pkg/tests/schema_pulumi_test.go [hunk 5]
```diff
@@ -4199,3 +4199,127 @@ outputs:
...
+		resUp := pt.Up()
+		// assert that the property gets resolved
+		require.Equal(t,
+			[]interface{}{"aux"},
+			resUp.Outputs["testOut"].Value,
+		)
+	}
+
+	t.Run("PRC enabled", func(t *testing.T) {
+		// TODO[pulumi/pulumi-terraform-bridge#2428]: Incorrect detailed diff with unknown elements
```
Comment (t0yv0) [inline pkg/tests/schema_pulumi_test.go:4283]: THank you!

File: pkg/tfbridge/diff_test.go [hunk 6]
```diff
@@ -1539,6 +1544,8 @@ func TestNestedComputedSetIntUpdateReplace(t *testing.T) {
 }
 
 func TestNestedComputedSetIntAdd(t *testing.T) {
+	// TODO[pulumi/pulumi-terraform-bridge#2428]: Incorrect detailed diff with unknown elements
+	t.Skip("Skipping until pulumi/pulumi-terraform-bridge#2428 is resolved")
```
Comment (t0yv0) [inline pkg/tfbridge/diff_test.go:1548]: 2428 is https://github.com/pulumi/pulumi-terraform-bridge/issues/2428 this is the wrong number perhaps?

**Review by t0yv0 — APPROVED** [review]
Looks good, thank you so much!

**Pull request merged**